tsconfig.json

```diff
@@ -2,7 +2,6 @@
   "compilerOptions": {
     "target": "esnext",
     "strict": true,
-    "lib": ["es2023"],
```

Comment:
This line seems to be problematic with my changes and old TS versions. Zustand supports the same range of TypeScript versions but lacks this line. Maybe it can be safely removed?

Reply:
It's added in #930. I think it's related to `} catch {` change.
If omitting this line goes well, I think it's fine to remove.

Reply:
Seems like using `"lib": ["dom"]` satisfies all TS versions.